pack.go

```diff
@@ -34,52 +34,102 @@ const (
-// AnnotationArtifactCreated or AnnotationCreated is provided, but its value
-// is not in RFC 3339 format.
-// Reference: https://www.rfc-editor.org/rfc/rfc3339#section-5.6
-var ErrInvalidDateTimeFormat = errors.New("invalid date and time format")
+// PackManifestType represents the manifest type used for [Pack].
+type PackManifestType = int32
+
+const (
+	// PackManifestTypeImageManifestLegacy represents the OCI Image Manifest
+	// type defined in image-spec v1.1.0-rc2.
+	// Reference: https://github.com/opencontainers/image-spec/blob/v1.1.0-rc2/manifest.md
+	PackManifestTypeImageManifestLegacy PackManifestType = iota
+
+	// PackManifestTypeImageManifest represents the OCI Image Manifest type
+	// defined since image-spec v1.1.0-rc4.
+	// Reference: https://github.com/opencontainers/image-spec/blob/v1.1.0-rc4/manifest.md
+	PackManifestTypeImageManifest
+)
 
-// PackOptions contains parameters for [oras.Pack].
+var (
+	// ErrInvalidDateTimeFormat is returned by [Pack] when
+	// AnnotationArtifactCreated or AnnotationCreated is provided, but its value
+	// is not in RFC 3339 format.
+	// Reference: https://www.rfc-editor.org/rfc/rfc3339#section-5.6
+	ErrInvalidDateTimeFormat = errors.New("invalid date and time format")
+
+	// ErrMissingArtifactType is returned by [Pack] when artifactType is not
+	// specified and the config media type is set to
+	// "application/vnd.oci.empty.v1+json" (the default config).
+	ErrMissingArtifactType = errors.New("missing artifact type")
+)
+
+// PackOptions contains parameters for [Pack].
 type PackOptions struct {
 	// Subject is the subject of the manifest.
 	Subject *ocispec.Descriptor
+
 	// ManifestAnnotations is the annotation map of the manifest.
 	ManifestAnnotations map[string]string
 
 	// PackImageManifest controls whether to pack an image manifest or not.
-	//   - If true, pack an image manifest; artifactType will be used as the
-	// the config descriptor mediaType of the image manifest.
-	//   - If false, pack an artifact manifest.
+	//   - If true, pack an OCI Image Manifest
+	//   - If false, pack an OCI Artifact Manifest
 	// Default: false.
 	PackImageManifest bool
+
+	// PackManifestType controls which type of manifest to pack.
+	// This option is valid only when PackImageManifest is true.
+	// Default: PackManifestTypeImageManifestLegacy.
+	PackManifestType
+
 	// ConfigDescriptor is a pointer to the descriptor of the config blob.
 	// If not nil, artifactType will be implied by the mediaType of the
 	// specified ConfigDescriptor, and ConfigAnnotations will be ignored.
 	// This option is valid only when PackImageManifest is true.
 	ConfigDescriptor *ocispec.Descriptor
+
 	// ConfigAnnotations is the annotation map of the config descriptor.
 	// This option is valid only when PackImageManifest is true
 	// and ConfigDescriptor is nil.
 	ConfigAnnotations map[string]string
 }
 
+// DefaultPackOptions provides the default PackOptions.
+var DefaultPackOptions PackOptions = PackOptions{
```

Comment:
You may document that the default pack options are subject to change in the future.